src/backend/routers/conversation.py

```diff
@@ -59,21 +59,33 @@ async def get_conversation(
 
 @router.get("", response_model=list[ConversationWithoutMessages])
 async def list_conversations(
-    *, offset: int = 0, limit: int = 100, session: DBSessionDep, request: Request
+    *,
+    offset: int = 0,
+    limit: int = 100,
+    agent_id: str = None,
+    session: DBSessionDep,
+    request: Request,
 ) -> list[ConversationWithoutMessages]:
     """
     List all conversations.
 
     Args:
         offset (int): Offset to start the list.
         limit (int): Limit of conversations to be listed.
+        agent_id (str): Query parameter for agent ID to optionally filter conversations by agent.
         session (DBSessionDep): Database session.
         request (Request): Request object.
 
     Returns:
         list[ConversationWithoutMessages]: List of conversations.
     """
     user_id = get_header_user_id(request)
+
+    if agent_id:
```

Comment:
Now don't need to conditional here, just pass `agent_id` directly and the crud method will handle it